src/plots/cartesian/set_convert.js

```diff
@@ -313,6 +313,14 @@ module.exports = function setConvert(ax) {
             return c === -1 ? BADNUM : c;
         };
 
+        ax.d2l_noadd = function(v) {
+            // d2c variant that that won't add categories but will also
+            // allow numbers to be mapped to the linearized axis positions
+            var index = ax._categories.indexOf(v);
+            if(index !== -1) return index;
+            if(typeof v === 'number') return v;
+        };
```

Comment:
before this change, sometimes tickvals could add categories to the list, but it happened after autorange calculations so they didn't show up (until you zoom out on the axis). It was weird. Now that can't happen. At some point we should probably look through all other uses of `ax.d2l` (or `ax.d2c` and `ax.d2r`, which are all the same function for category axes) and make sure they shouldn't also be using `d2l_noadd`... but so far I haven't seen any other ill effects.